demos/common/models/src/detection_model_retinaface.cpp

```diff
@@ -0,0 +1,391 @@
+        masks->push_back(memPtr[start + offset]);
+    }
+}
+
+std::vector<int> nms(const std::vector<ModelRetinaFace::Anchor>& boxes, const std::vector<float>& scores, const float thresh) {
+    std::vector<float> areas(boxes.size());
+    for (int i = 0; i < boxes.size(); ++i) {
+        areas[i] = (boxes[i].right - boxes[i].left) * (boxes[i].bottom - boxes[i].top);
+    }
+    std::vector<int> order(scores.size());
+    std::iota(order.begin(), order.end(), 0);
+    std::sort(order.begin(), order.end(), [&scores](int o1, int o2) { return scores[o1] > scores[o2]; });
+
+    int ordersNum = 0;
+    for (; ordersNum < order.size() && scores[order[ordersNum]] >= 0; ordersNum++);
+
+    std::vector<int> keep;
+    bool shouldContinue = true;
+    for (int i = 0; shouldContinue && i < ordersNum; ++i) {
+        auto idx1 = order[i];
+        if (idx1 >= 0) {
+            keep.push_back(idx1);
+            shouldContinue = false;
+            for (int j = i + 1; j < ordersNum; ++j) {
+                auto idx2 = order[j];
+                if (idx2 >= 0) {
+                    shouldContinue = true;
+                    auto overlappingWidth = std::fminf(boxes[idx1].right, boxes[idx2].right) - std::fmaxf(boxes[idx1].left, boxes[idx2].left);
+                    auto overlappingHeight = std::fminf(boxes[idx1].bottom, boxes[idx2].bottom) - std::fmaxf(boxes[idx1].top, boxes[idx2].top);
+                    auto intersection = overlappingWidth > 0 && overlappingHeight > 0 ? overlappingWidth * overlappingHeight : 0;
+                    auto overlap = intersection / (areas[idx1] + areas[idx2] - intersection);
+
+                    if (overlap >= thresh) {
+                        order[j] = -1;
+                    }
+                }
+            }
+        }
+    }
+    return keep;
+}
+
+std::unique_ptr<ResultBase> ModelRetinaFace::postprocess(InferenceResult& infResult) {
+    std::vector<float> scores;
+    scores.reserve(INIT_VECTOR_SIZE);
+    std::vector<Anchor> bboxes;
+    bboxes.reserve(INIT_VECTOR_SIZE);
+    std::vector<cv::Point2f> landmarks;
+    landmarks.reserve(INIT_VECTOR_SIZE);
+    std::vector<float> masks;
+
+    if (shouldDetectMasks) {
+        masks.reserve(INIT_VECTOR_SIZE);
+    }
+
+// --------------------------- Gather & Filter output from all levels ----------------------------------------------------------
+    for (int idx = 0; idx < anchorCfg.size(); ++idx) {
+        const auto bboxRaw = infResult.outputsData[separateOutputsNames[OT_BBOX][idx]];
+        const auto scoresRaw = infResult.outputsData[separateOutputsNames[OT_SCORES][idx]];
+        const auto landmarksRaw = infResult.outputsData[separateOutputsNames[OT_LANDMARK][idx]];
+        auto sz = bboxRaw->getTensorDesc().getDims();
+        auto height = sz[2];
+        auto width = sz[3];
+        auto s = anchorCfg[idx].stride;
+        auto anchorNum = anchorsFpn[s].size();
+
+        //--- Creating strided anchors plane at first start
```

Comment:
Shouldn't this be in `prepareInputsOutputs()`? It looks it depends only on output shape of OT_BBOX.

Reply:
Done.